olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+
+#include <boost/optional.hpp>
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/ApiResponse.h>
+#include <olp/dataservice/read/ConsumerProperties.h>
+#include "generated/model/SubscribeResponse.h"
+
+namespace olp {
+namespace client {
+class OlpClient;
+class CancellationContext;
+}  // namespace client
+
+namespace dataservice {
+namespace read {
+
+/// Response template type
+template <typename ResultType>
+using Response = client::ApiResponse<ResultType, client::ApiError>;
+
+/**
+ * @brief The `stream` service provides the ability to consume data from a
+ * stream layer. With this service you can subscribe to a stream layer and
+ * consume messages.
+ */
+class StreamApi {
+ public:
+  /// Subscribe response type.
+  using SubscribeApiResponse = Response<model::SubscribeResponse>;
+
+  /// Unsubscribe response type. Returns status of the HTTP request on success.
+  using UnsubscribeApiResponse = Response<int>;
+
+  /**
+   * @brief Enable message consumption from a specific stream layer.
+   *
+   * Use the base path returned from the API Lookup service.\nNote: For **mode =
+   * parallel**, one unit of parallelism currently equals 1 MBps inbound or 2
+   * MBps outbound, whichever is greater, rounded up to the nearest integer. The
+   * number of subscriptions within the same group cannot exceed the parallelism
+   * allowed. For more details see [Get Data from a Stream
+   * Layer](https://developer.here.com/olp/documentation/data-store/data_dev_guide/rest/getting-data-stream.html).
+   *
+   * @param client Instance of OlpClient used to make REST request.
+   * @param layer_id The ID of the stream layer.
+   * @param subscription_id Include this parameter if you want to look up the
+   * `nodeBaseURL` for a given subscriptionId.
+   * @param mode Specifies whether to use serial or parallel subscription mode.
+   * For more details see [Get Data from a Stream
+   * Layer](https://developer.here.com/olp/documentation/data-store/data_dev_guide/rest/getting-data-stream.html).
```

Comment:
I think better to delete this sentence - no need to repeat the same link that you stated in the detailed description.